src/libpmemset/pmemset.c

```diff
@@ -187,12 +213,31 @@ pmemset_part_map(struct pmemset_part **part, struct pmemset_extras *extra,
 	struct pmemset_part *p = *part;
 	struct pmemset_part_map *part_map;
 	struct pmemset *set = pmemset_part_get_pmemset(p);
+	struct pmemset_config *set_config = pmemset_get_pmemset_config(set);
+	enum pmem2_granularity mapping_gran;
+	enum pmem2_granularity config_gran =
+		pmemset_get_config_granularity(set_config);
 
 	int ret = pmemset_part_create_part_mapping(&part_map, p,
-			PMEM2_GRANULARITY_PAGE);
+			config_gran, &mapping_gran);
 	if (ret)
 		return ret;
 
+	/*
+	 * effective granularity is only set once and
+	 * must have the same value for each mapping
+	 */
+	enum pmem2_granularity set_effective_gran =
+		pmemset_get_store_granularity(set);
+	if (set_effective_gran == PMEMSET_GRANULARITY_INVALID) {
+		pmemset_set_store_granularity(set, mapping_gran);
+	} else if (set_effective_gran != mapping_gran) {
+		ERR("all parts in the set must have the same granularity %d",
```

Comment:
the part granularity is ..., all parts must have the same granularity ...

(use strings, not numbers)